noir-projects/aztec-nr/aztec/src/macros/notes/mod.nr

```diff
@@ -500,30 +536,70 @@ comptime fn get_setup_log_plaintext_body(
 /// Example:
 /// ```
 /// struct TokenNoteFinalizationPayload {
-///     log: [Field; 2],
-///     note_hash: Field
+///     context: &mut aztec::prelude::PublicContext,
+///     hiding_point_slot: Field,
+///     setup_log_slot: Field,
+///     public_values: [Field; 2]
```

Comment:
For the finalization payload I decided to move context into it because that allows me to fetch the data inside the macro generated struct. This made sense to do because then I can populate the log field lengths and byte lenghts at comptime and dev does not need to deal with that.